docs/source/commandline/commandline-all.rst

```diff
@@ -346,81 +678,244 @@ OSD-related Options
+      - auto [#MIPAutoWindows]_.
+      - rawinput
+      - dinput
+      - win32
+      - none
+      - sdl [#MIPSDLWindows]_.
+    * - **SDL (macOS and Linux)**
+      - auto [#MIPAutoSDL]_.
+      - 
+      - 
+      - 
+      - none
+      - sdl
+    * - **Linux**
+      - auto [#MIPAutoSDL]_.
+      - 
+      - 
+      - 
+      - none
+      - sdl
+
+..  rubric:: Footnotes
+
+..  [#MIPAutoWindows] On Windows, ``auto`` will try ``rawinput`` with fallback
+                      to ``dinput``.
+
+..  [#MIPSDLWindows] SDL support on Windows requires that you compile MAME with
+                      the support in. By default SDL is not included in Windows
+                      builds of MAME.
+
+..  [#MIPAutoSDL] ``auto`` on SDL will default to ``sdl``.
 
-| On Windows, you can choose from: ``auto``, ``rawinput``, ``dinput``, ``win32``, or ``none``
-| On SDL, you can choose from: ``auto``, ``sdl``, ``none``
-|
-| The default is ``auto``.
-|
-| On Windows, ``auto`` will try ``rawinput`` with fallback to ``dinput``.
-| On SDL, ``auto`` will default to ``sdl``.
-|
+Example:
+    .. code-block:: bash
+
+        mame64 indy_4610 -mouseprovider win32
 
 .. _mame-commandline-lightgunprovider:
 
 **\-lightgunprovider**
 
-    Chooses how MAME will get light gun input.
+    Chooses how MAME will get light gun input. The default is ``auto``.
+
+.. list-table:: Supported light gun input providers per-platform
+    :header-rows: 0
+    :stub-columns: 0
+
+    * - **Microsoft Windows**
+      - auto [#LGIPAutoWindows]_.
+      - rawinput
+      - win32
+      - none
+      - 
+      - 
+    * - **macOS**
+      - auto [#LGIPAutoSDL]_.
+      - 
+      - 
+      - none
+      - 
+      - 
+    * - **Linux**
+      - auto [#LGIPAutoLinux]_.
+      - 
+      - 
+      - none
+      - 
+      - x11
+
+..  rubric:: Footnotes
+
+..  [#LGIPAutoWindows] On Windows, auto will try ``rawinput`` with fallback to
+                       ``win32``, or ``none`` if it doesn't find any.
+
+..  [#LGIPAutoSDL] On non-Linux SDL, ``auto`` will default to ``none``.
+
+..  [#LGIPAutoLinux] On SDL/Linux, ``auto`` will default to ``x11``, or ``none``
+                     if it doesn't find any.
+
+Example:
+    .. code-block:: bash
 
-| On Windows, you can choose from: ``auto``, ``rawinput``, ``win32``, or ``none``
-| On SDL, you can choose from: ``auto``, ``x11`` or ``none``
-|
-| The default is ``auto``.
-|
-| On Windows, auto will try ``rawinput`` with fallback to ``win32``, or ``none`` if it doesn't find any.
-| On SDL/Linux, ``auto`` will default to ``x11``, or ``none`` if it doesn't find any.
-| On other SDL, ``auto`` will default to ``none``.
-|
+        mame64 lethalen -lightgunprovider x11
 
 .. _mame-commandline-joystickprovider:
 
 **\-joystickprovider**
 
-    Chooses how MAME will get joystick input.
+    Chooses how MAME will get joystick input. The default is ``auto``.
+
+.. list-table:: Supported light gun input providers per-platform
```

Comment:
This should say “joystick”, not “light gun”.

Reply:
Not sure how I missed that despite rereading it three times. Corrected.